substratest/assets.py

```diff
@@ -88,21 +89,24 @@ class _BaseFutureMixin(abc.ABC):
 
     def attach(self, session):
         """Attach session to asset."""
-        self._session = session
+        self.__session = session
         return self
 
+    @property
+    def _session(self):
+        if not self.__session:
+            raise errors.TError(f'No session attached with {self}')
+        return self.__session
```

Comment:
This is very nice. I originally didn't want to directly use the attached session because it could not have been there, but this makes it very neat and explicit. 👍